### PR TITLE
fix: set correct permision UI during joining the community

### PR DESCRIPTION
**Pull request by mprakhov**
### What does the PR do

Fixes
- crash fix
- set correct permission required UI during joining the closed community
- if community permission is closed and the user owns TM or admin token - show this token in the join community permission view
- if all channels are hidden, show the "You don't hold the necessary tokens" UI only if the community is not closed and we fit the permissions to join the community
- do not show join community permission view if it is open community

Closes: 
- https://github.com/status-im/status-desktop/issues/14091
- https://github.com/status-im/status-desktop/issues/14085
- https://github.com/status-im/status-desktop/issues/14104

IMPACT:
- join community permission view for a member, who wants to join
- showing community channels permission

### CODE REVIEW

**Comment by anastasiyaig** [comment]
@mprakhov what is correct UI here? Any figma link?

**Comment by anastasiyaig** [comment]
Meanwhile i launched our community autotests and they are all good https://ci.status.im/job/status-desktop/job/e2e/job/manual/1705/allure/#suites/a9b9ce4ef70730fb5814a5009e98b445

@glitchminer maybe it could reduce some work for you

**Review by MishkaRogachev — APPROVED** [review]

File: ui/app/AppLayouts/Chat/views/ChatView.qml [hunk 1]
```diff
@@ -13,6 +13,7 @@ import shared.stores.send 1.0
 import SortFilterProxyModel 0.2
 
 import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/views/ChatView.qml:16]: Is it needed?

File: ui/imports/shared/stores/PermissionsStore.qml [hunk 2]
```diff
@@ -97,7 +97,9 @@ QtObject {
         id: becomeMemberPermissionsModel
         sourceModel: root.permissionsModel
         function filterPredicate(modelData) {
-            return (modelData.permissionType == Constants.permissionType.member) &&
+            return (modelData.permissionType === Constants.permissionType.member ||
+                    modelData.permissionType === Constants.permissionType.admin ||
```
Comment (noeliaSD) [inline ui/imports/shared/stores/PermissionsStore.qml:101]: Why the `admin` role is added here if the model is about becoming member / joining community??

**Review by jrainville — APPROVED** [review]
Looks good

**Comment by jrainville** [comment]
This will be needed in 2.28 after being tested and merged

**Comment by noeliaSD** [comment]
> @anastasiyaig , @glitchminer waiting for other UI fixes from Noelia (she will merge them to this PR) and I think tomorrow this PR can be tested

I'll ping you when ready!

File: storybook/pages/JoinCommunityPermissionsEditor.qml [hunk 3]
```diff
@@ -125,6 +125,16 @@ ColumnLayout {
             text: "Long model"
             onCheckedChanged: if(checked) root.communityHoldingsModel = PermissionsModel.longPermissionsModel
         }
+
+        RadioButton {
+            text: "Private met model"
```
Comment (kounkou) [inline storybook/pages/JoinCommunityPermissionsEditor.qml:130]: I am not sure what is a `Private met model` ?

**Comment by glitchminer** [comment]
Hi @mprakhov , noted a couple of possible issues:

1. When the permissions of a channel are changed to hidden then the chat goes from the blurred permissions screen to being visible

https://github.com/status-im/status-desktop/assets/50769329/096737ae-a73f-4a0f-a5fa-18fdba1ae943

---
2. In the case where both "view and post" and "view" permissions are configured on a channel. If a member meets the requirements for "view and post" permissions but not "view" permissions then the user shown the chat and is told they must hold tokens that they already have if they want to post. Likely to be an edge case? 
image
image

---

Other stuff:
All community membership permissions hidden worked OK
Permissions screen not shown if no membership permissions set OK
Meeting requirements to be admin or TM while not meeting permissions to be a member worked OK

**Review by kounkou — APPROVED** [review]
Changes look good to me. 🚢  it!

File: storybook/pages/JoinCommunityPermissionsEditor.qml (hunk 3 above)
Comment (noeliaSD) [reply storybook/pages/JoinCommunityPermissionsEditor.qml:130]: It means a model where all the permission rows are private and at least one is met

**Comment by anastasiyaig** [comment]
thanks @mprakhov and @glitchminer !

**Pull request closed without merging**